src/vs/workbench/contrib/webview/browser/webviewElement.ts

```diff
@@ -103,7 +108,7 @@ export class WebviewElement extends Disposable implements IWebview, WebviewFindD
 		if (!this._focused) {
 			return false;
 		}
-		if ($window.document.activeElement && $window.document.activeElement !== this.element) {
+		if (this.codeWindow.document.activeElement && this.codeWindow.document.activeElement !== this.element) {
```

Comment:
Instead of using `codeWindow`, could we get the window using `this.element.ownerDocument` here?

Reply:
@mjbvz the reason we passed in `codeWindow` (initially for notebook) is `element.onwerDocument` is always `mainWindow` before it's attached to the grid view. That means if we create the editor/webview, initilize it and then lastly attach it to the DOM tree, we would be reading the wrong `codeWindow` in our initialization code. That's while making this work for Notebooks, we intentioally compute what's the right target `codeWindow` and enforce the webview to use it. Maybe it has change

Reply:
@bpasero can probably also answer this.

Reply:
Yeah, this is an unfortunate consequence of the `Grid` widget working different than other widgets: instead of passing in the `HTMLElement` it should be contained in, you created it first and then `appendChild` it to the right container. That means, any part of the grid (editors) that are created will get the "wrong" main window when they use `element.ownerDocument` until the grid has been appended into the aux window.

I sat down today with Joao for 1h to see if we can change this, but we found it to be quite complex. We may have a fix eventually but for now have to pass the `CodeWindow` around in places where we cannot get it otherwise.